chain/network/src/routing.rs

```diff
@@ -367,6 +387,94 @@ impl RoutingTable {
         })
     }
 
+    /// Get the nonce of the component where the peer was stored
+    fn component_nonce_from_peer(&mut self, peer_id: PeerId) -> Result<u64, ()> {
+        match self.store.get_ser::<u64>(ColPeerComponent, Vec::from(peer_id).as_ref()) {
+            Ok(Some(nonce)) => Ok(nonce),
+            _ => Err(()),
+        }
+    }
+
+    /// Get all edges in the component with `nonce`
+    /// Remove those edges from the store.
+    fn get_component_edges(
+        &mut self,
+        nonce: u64,
+        update: &mut StoreUpdate,
+    ) -> Result<Vec<Edge>, ()> {
+        let enc_nonce = index_to_bytes(nonce);
+
+        let result = match self.store.get_ser::<Vec<Edge>>(ColComponentEdges, enc_nonce.as_ref()) {
+            Ok(Some(edges)) => Ok(edges),
+            _ => Err(()),
+        };
+
+        update.delete(ColComponentEdges, enc_nonce.as_ref());
+
+        result
+    }
+
+    /// If peer_id is not on memory check if it is on disk in bring it back on memory.
+    fn touch(&mut self, peer_id: &PeerId) {
+        if peer_id == self.peer_id() || self.peer_last_time_reachable.contains_key(peer_id) {
+            return;
+        }
+
+        let me = self.peer_id().clone();
+
+        if let Ok(nonce) = self.component_nonce_from_peer(peer_id.clone()) {
+            let mut update = self.store.store_update();
+
+            if let Ok(edges) = self.get_component_edges(nonce, &mut update) {
+                for edge in edges.into_iter() {
```

Comment:
`into_iter` is not necessary if you want to consume the variable

Reply:
My understanding is exactly the opposite, if you want to consume the variables  use `into_iter` and and get ownership of items, otherwise, use `iter` and get reference to items. In this case I don't care about the iterator anymore, so I'm getting ownership of each item.

Reply:
If you take ownership you don't need into_iter, you can experiment with this. After you do a loop without `into_iter` you will not be able to access the object again.

Reply:
ok, I just learnt how `for` works on rust, we don't need explict `iter` or `into_iter` always.